### PR TITLE
Session name

### PR DESCRIPTION
**Pull request by AlbertoCasasOrtiz**
To be merged with https://github.com/stanfordnmbl/opencap-api/pull/67. 
Migration not required (I added it to metadata).

### CODE REVIEW

File: src/components/pages/SelectSession.vue [hunk 1]
```diff
@@ -64,6 +64,50 @@
...
+                      <v-row class="m-0">
+                        <v-col cols="2">
+                          <v-icon x-large color="orange">mdi-rename-box</v-icon>
+                        </v-col>
+                        <v-col cols="10">
+                          <p class="mb-1">
+                            Insert a new name for session {{item.sessionName}}:
+                          </p>
+                          <small class="mt-0">
+                            Only alphanumeric characters and underscore (_) are allowed.
```
Comment (antoinefalisse) [inline src/components/pages/SelectSession.vue:87]: Rephrase: Only alphanumeric characters and underscores are allowed.

File: src/components/pages/Step4.vue [hunk 2]
```diff
@@ -588,7 +606,8 @@ export default {
                   // subject_gender: this.gender,
                   settings_data_sharing: this.data_sharing,
                   settings_pose_model: this.pose_model,
-                  settings_framerate: this.framerate
+                  settings_framerate: this.framerate,
+                  sessionName: this.sessionName
```
Comment (antoinefalisse) [inline src/components/pages/Step4.vue:610]: Let's follow the same conventions: `settings_session_name`

**Review by antoinefalisse — CHANGES_REQUESTED** [review]
Thanks @AlbertoCasasOrtiz, I left a few minor comments. Some questions too:

1. Is the session name by default the session ID? (I have not tested yet).
2. If users rename their session, how do they access the session ID? opencap-core and opencap-processing currently rely on session IDs for processing data. This PR should therefore come together with PRs on these repos to support entering session names or session IDs. I can help with that.

Is there a way you record a quick video that show how it works? It is easier than merging and checking for me (I should have a end-to-end local pipeline I know), thanks.

**Comment by antoinefalisse** [comment]
Another comment I have. I have not tested so maybe it is already implemented, but it might be incomplete to only allow renaming sessions and not also naming sessions in the first place. Users might want to name the session when starting a data collection, and I feel that this PR only allows renaming after the fact. Is that the case? If so, we should think about how to allow users to name the session as part of the workflow. Maybe when starting a session, a pop up window should appear asking to name the session, and then you get to the screen with the QR code. Thoughts @AlbertoCasasOrtiz  and @suhlrich?

**Comment by suhlrich** [comment]
I tested locally. Rename works, but the date format gets changed when I rename: 
image

**Comment by suhlrich** [comment]
It looks like you can create a session name in Step 4. Can we make the session and subject boxes smaller, it'd be nice not to have to scroll to do everything. Let's have the subject box on top, then have a box with no title that has 'Session Name (optional)' as the default.

image

File: src/components/pages/Step4.vue (hunk 2 above)
Comment (AlbertoCasasOrtiz) [reply src/components/pages/Step4.vue:610]: Applied.

**Comment by AlbertoCasasOrtiz** [comment]
Hi, I answer some questions here:

1. The default name right now is an empty string. The session name is in the metadata field, so I didn't had to touch how the UUIDs work, everything else should work as before.
2. In the tables through the application, both IDs and names are shown, so users can identify sessions by both, sessions and names.
3. You can name sessions when you create them in Step 4. I added a field for this. I added this to step 4 and not at the beginning, so users only have to insert session name when the calibration and everything is done.
4. Regarding the date change, we discovered that bug during the dev demos. I just fixed it.
5. I have applied the changes to step 4, so the session name box do not occupy too much space, and (optional) to the title.

**Comment by suhlrich** [comment]
awesome! Can you post a screenshot of the new step 4?

**Comment by AlbertoCasasOrtiz** [comment]
Sure, here it is. I am still getting a vertical scrollbar because of the images box at right.

Maybe we can save more space if we put session name and subject in the same box directly? In that case, instead of select subject,  the title of the box could be "Session settings" or something like that.

screenshot

**Comment by suhlrich** [comment]
I like that idea. Let's have one box "Session Info" with subjects dropdown and the optional session name.

**Comment by AlbertoCasasOrtiz** [comment]
Done

**Pull request merged**